database/datastore/super-mario-maker/get_latest_courses_with_limit.go

```diff
@@ -0,0 +1,116 @@
+package datastore_smm_db
+
+import (
+	"database/sql"
+	"time"
+
+	"github.com/PretendoNetwork/nex-go"
+	datastore_super_mario_maker_types "github.com/PretendoNetwork/nex-protocols-go/datastore/super-mario-maker/types"
+	datastore_types "github.com/PretendoNetwork/nex-protocols-go/datastore/types"
+	"github.com/PretendoNetwork/super-mario-maker-secure/database"
+	datastore_db "github.com/PretendoNetwork/super-mario-maker-secure/database/datastore"
+	"github.com/PretendoNetwork/super-mario-maker-secure/globals"
+	"github.com/lib/pq"
+)
+
+func GetLatestCoursesWithLimit(limit int) ([]*datastore_super_mario_maker_types.DataStoreCustomRankingResult, uint32) {
+	courses := make([]*datastore_super_mario_maker_types.DataStoreCustomRankingResult, 0)
+
+	rows, err := database.Postgres.Query(`
+		SELECT
+			object.data_id,
+			object.owner,
+			object.size,
+			object.name,
+			object.data_type,
+			object.meta_binary,
+			object.permission,
+			object.permission_recipients,
+			object.delete_permission,
+			object.delete_permission_recipients,
+			object.period,
+			object.refer_data_id,
+			object.flag,
+			object.tags,
+			object.creation_date,
+			object.update_date,
+			ranking.value
+		FROM datastore.objects object
+		JOIN datastore.object_custom_rankings ranking
+		ON
+			object.data_id = ranking.data_id AND
+			object.upload_completed = TRUE AND
+			object.deleted = FALSE AND
+			object.under_review = FALSE AND
+			ranking.application_id = 0
+		ORDER BY object.creation_date DESC
```

Comment:
This function is the same as random courses but the query is ordered by creation date, *descending* (so the newest courses appear first)